corehq/messaging/smsbackends/telerivet/views.py

```diff
@@ -60,14 +60,21 @@ def message_status(request, message_id):
     status = request.POST.get('status')
     logger.info(f'Updating Telerivet message status: message_id={message_id}, status={status}')
 
-    process_message_status.delay(
-        message_id,
-        status,
-        request.POST.get('secret', ''),
-        error_message=request.POST.get('error_message', ''),
-    )
+    backend = SQLTelerivetBackend.by_webhook_secret(request.POST.get('secret', ''))
+    if backend is None:
+        return HttpResponseForbidden()
 
-    return HttpResponse()
+    try:
+        sms = SMS.objects.get(couch_id=message_id)
+        process_message_status(
+            sms,
+            status,
+            error_message=request.POST.get('error_message', ''),
+        )
+        return HttpResponse()
+
+    except SMS.DoesNotExist:
+        raise Http404()
```

Comment:
nit: 404. Just curious to why 404?

Reply:
I thought either 404 (since the url contains a message id by which to search for message) or 400, but I went with the 404 for no special reason really...although I'm not set on 404. What is your thoughts?

Reply:
I have good with either. 
If we can confirm what HQ API returns if we send a missing ID for user api end points for example, the behavior will be consistent.

Reply:
[Here's an example](https://github.com/dimagi/commcare-hq/blob/40da81ece0bdece921c1b186ff45838562a7b024/corehq/apps/hqadmin/views/users.py#L166) of where a user found by username does not exist and 404 returned, so I guess sticking to 404 is best then.